File: androidApp/src/main/java/com/cocktailcraft/screens/ProfileScreen.kt
package com.cocktailcraft.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.DarkMode
import androidx.compose.material.icons.filled.DateRange
import androidx.compose.material.icons.filled.Email
import androidx.compose.material.icons.filled.ExitToApp
import androidx.compose.material.icons.filled.Help
import androidx.compose.material.icons.filled.LightMode
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Notifications
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.VisibilityOff
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.cocktailcraft.navigation.NavigationManager
import com.cocktailcraft.ui.theme.AppColors
import com.cocktailcraft.viewmodel.ProfileViewModel
import com.cocktailcraft.viewmodel.ThemeViewModel

@Composable
fun ProfileScreen(
    navigationManager: NavigationManager,
    profileViewModel: ProfileViewModel = viewModel(),
    themeViewModel: ThemeViewModel = viewModel()
) {
    val context = LocalContext.current

    // Get user data from ViewModel
    val user by profileViewModel.user.collectAsState()
    val isSignedIn by profileViewModel.isSignedIn.collectAsState()
    val isLoading by profileViewModel.isLoading.collectAsState()
    val error by profileViewModel.error.collectAsState()

    // Get theme data from ThemeViewModel
    val isDarkMode by themeViewModel.isDarkMode.collectAsState()

    // Dialog states
    var showLogoutDialog by remember { mutableStateOf(false) }
    var showSignInDialog by remember { mutableStateOf(false) }
    var showSignUpDialog by remember { mutableStateOf(false) }

    // Sample profile data (in a real app, this would come from a ViewModel)
    val userName = user?.name ?: "Guest User"
    val userEmail = user?.email ?: "guest@example.com"

    // Sign in dialog state
    if (showSignInDialog) {
        SignInDialog(
            onDismiss = { showSignInDialog = false },
            onSignIn = { email, password ->
                profileViewModel.signIn(email, password)
                showSignInDialog = false
            }
        )
    }

    // Sign up dialog state
    if (showSignUpDialog) {
        SignUpDialog(
            onDismiss = { showSignUpDialog = false },
            onSignUp = { name, email, password ->
                profileViewModel.signUp(name, email, password)
                showSignUpDialog = false
            }
        )
    }

    // Logout dialog state
    if (showLogoutDialog) {
        AlertDialog(
            onDismissRequest = { showLogoutDialog = false },
            title = { Text("Logout") },
            text = { Text("Are you sure you want to logout?") },
            confirmButton = {
                Button(
                    onClick = {
                        profileViewModel.signOut()
                        showLogoutDialog = false
                        // Navigate to Home screen after logout
                        navigationManager.navigateToHome()
                    },
                    colors = ButtonDefaults.buttonColors(
                        containerColor = AppColors.Primary
                    )
                ) {
                    Text("Logout")
                }
            },
            dismissButton = {
                OutlinedButton(
                    onClick = { showLogoutDialog = false }
                ) {
                    Text("Cancel")
                }
            }
        )
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(AppColors.Background)
            .verticalScroll(rememberScrollState())
            .padding(16.dp)
    ) {
        // Profile header
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(
                containerColor = AppColors.Surface
            ),
            elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
        ) {
            Column(
                modifier = Modifier
                    .padding(16.dp)
                    .fillMaxWidth(),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                // Profile picture
                Box(
                    modifier = Modifier
                        .size(80.dp)
                        .background(AppColors.Primary.copy(alpha = 0.2f), CircleShape),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = userName.take(1).uppercase(),
                        fontSize = 32.sp,
                        fontWeight = FontWeight.Bold,
                        color = AppColors.Primary
                    )
                }

                Spacer(modifier = Modifier.height(16.dp))

                // User name
                Text(
                    text = userName,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.TextPrimary
                )

                Spacer(modifier = Modifier.height(4.dp))

                // User email
                Text(
                    text = userEmail,
                    fontSize = 14.sp,
                    color = AppColors.TextSecondary
                )

                // Show login/signup buttons if not signed in
                if (!isSignedIn) {
                    Spacer(modifier = Modifier.height(24.dp))

                    Text(
                        text = "Sign in to access your profile",
                        fontSize = 16.sp,
                        color = AppColors.TextSecondary,
                        modifier = Modifier.padding(bottom = 16.dp)
                    )

                    Button(
                        onClick = { showSignInDialog = true },
                        colors = ButtonDefaults.buttonColors(
                            containerColor = AppColors.Primary
                        ),
                        modifier = Modifier
                            .fillMaxWidth(0.8f)
                            .padding(vertical = 4.dp)
                    ) {
                        Text("Sign In")
                    }

                    Spacer(modifier = Modifier.height(8.dp))

                    OutlinedButton(
                        onClick = { showSignUpDialog = true },
                        modifier = Modifier
                            .fillMaxWidth(0.8f)
                            .padding(vertical = 4.dp)
                    ) {
                        Text("Create Account")
                    }
                }
            }
        }

        // Only show account settings and logout option if signed in
        if (isSignedIn) {
            // Account settings
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 16.dp),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(
                    containerColor = AppColors.Surface
                ),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
            ) {
                Column(
                    modifier = Modifier.padding(16.dp)
                ) {
                    Text(
                        text = "Account Settings",
                        fontSize = 18.sp,
                        fontWeight = FontWeight.Bold,
                        color = AppColors.TextPrimary,
                        modifier = Modifier.padding(bottom = 16.dp)
                    )

                    SettingsItem(
                        icon = Icons.Default.Person,
                        title = "Edit Profile",
                        onClick = { /* Handle edit profile */ }
                    )

                    SettingsItem(
                        icon = Icons.Default.Lock,
                        title = "Change Password",
                        onClick = { /* Handle change password */ }
                    )

                    SettingsItem(
                        icon = Icons.Default.Email,
                        title = "Email Preferences",
                        onClick = { /* Handle email preferences */ }
                    )

                    SettingsItem(
                        icon = Icons.Default.Notifications,
                        title = "Notification Settings",
                        onClick = { /* Handle notification settings */ }
                    )
                }
            }
        }

        // App settings
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(
                containerColor = AppColors.Surface
            ),
            elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
        ) {
            Column(
                modifier = Modifier.padding(16.dp)
            ) {
                Text(
                    text = "App Settings",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.TextPrimary,
                    modifier = Modifier.padding(bottom = 16.dp)
                )

                SettingsItem(
                    icon = Icons.Default.DateRange,
                    title = "Order History",
                    onClick = { navigationManager.navigateToOrderList() }
                )

                SettingsItem(
                    icon = Icons.Default.Help,
                    title = "Help & Support",
                    onClick = { /* Handle help & support */ }
                )

                // Dark Mode Toggle with animation
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable { themeViewModel.toggleDarkMode() }
                        .padding(vertical = 12.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    // Animated icon that changes based on the theme
                    val iconTint = if (isDarkMode) AppColors.PrimaryDark else AppColors.TextPrimary

                    Icon(
                        imageVector = if (isDarkMode) Icons.Default.DarkMode else Icons.Default.LightMode,
                        contentDescription = if (isDarkMode) "Switch to Light Mode" else "Switch to Dark Mode",
                        tint = iconTint,
                        modifier = Modifier.size(24.dp)
                    )

                    Spacer(modifier = Modifier.width(16.dp))

                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            text = "Dark Mode",
                            fontSize = 16.sp,
                            color = AppColors.TextPrimary
                        )

                        Text(
                            text = if (isDarkMode) "On" else "Off",
                            fontSize = 12.sp,
                            color = AppColors.TextSecondary
                        )
                    }

                    Switch(
                        checked = isDarkMode,
                        onCheckedChange = { themeViewModel.toggleDarkMode() }
                    )
                }

                // Only show logout if signed in
                if (isSignedIn) {
                    SettingsItem(
                        icon = Icons.Default.ExitToApp,
                        title = "Logout",
                        onClick = { showLogoutDialog = true },
                        textColor = AppColors.Error
                    )
                }
            }
        }

        // App information
        Card(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(
                containerColor = AppColors.Surface
            ),
            elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
        ) {
            Column(
                modifier = Modifier.padding(16.dp)
            ) {
                Text(
                    text = "About",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.TextPrimary,
                    modifier = Modifier.padding(bottom = 16.dp)
                )

                Text(
                    text = "Cocktail Bar App",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Medium,
                    color = AppColors.TextPrimary
                )

                Spacer(modifier = Modifier.height(4.dp))

                Text(
                    text = "Version 1.0.0",
                    fontSize = 14.sp,
                    color = AppColors.TextSecondary
                )

                Spacer(modifier = Modifier.height(16.dp))

                Text(
                    text = "© 2023 Cocktail Bar. All rights reserved.",
                    fontSize = 12.sp,
                    color = AppColors.TextSecondary
                )
            }
        }
    }

    // Show loading indicator if needed
    if (isLoading) {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(Color.Black.copy(alpha = 0.5f)),
            contentAlignment = Alignment.Center
        ) {
            CircularProgressIndicator(color = AppColors.Primary)
        }
    }

    // Show error message if needed
    error?.let { errorMessage ->
        AlertDialog(
            onDismissRequest = { profileViewModel.clearError() },
            title = { Text("Error") },
            text = { Text(errorMessage) },
            confirmButton = {
                Button(
                    onClick = { profileViewModel.clearError() },
                    colors = ButtonDefaults.buttonColors(
                        containerColor = AppColors.Primary
                    )
                ) {
                    Text("OK")
                }
            }
        )
    }
}

@Composable
fun SignInDialog(
    onDismiss: () -> Unit,
    onSignIn: (email: String, password: String) -> Unit
) {
    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var passwordVisible by remember { mutableStateOf(false) }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Sign In") },
        text = {
            Column {
                OutlinedTextField(
                    value = email,
                    onValueChange = { email = it },
                    label = { Text("Email") },
                    singleLine = true,
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 8.dp),
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email)
                )

                OutlinedTextField(
                    value = password,
                    onValueChange = { password = it },
                    label = { Text("Password") },
                    singleLine = true,
                    visualTransformation = if (passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 8.dp),
                    trailingIcon = {
                        IconButton(onClick = { passwordVisible = !passwordVisible }) {
                            Icon(
                                imageVector = if (passwordVisible) Icons.Default.Visibility else Icons.Default.VisibilityOff,
                                contentDescription = if (passwordVisible) "Hide password" else "Show password"
                            )
                        }
                    }
                )
            }
        },
        confirmButton = {
            Button(
                onClick = { onSignIn(email, password) },
                enabled = email.isNotBlank() && password.isNotBlank(),
                colors = ButtonDefaults.buttonColors(
                    containerColor = AppColors.Primary
                )
            ) {
                Text("Sign In")
            }
        },
        dismissButton = {
            OutlinedButton(onClick = onDismiss) {
                Text("Cancel")
            }
        }
    )
}

@Composable
fun SignUpDialog(
    onDismiss: () -> Unit,
    onSignUp: (name: String, email: String, password: String) -> Unit
) {
    var name by remember { mutableStateOf("") }
    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var passwordVisible by remember { mutableStateOf(false) }

    AlertDialog(
        onDismissRequest = onDismiss,
        title = { Text("Create Account") },
        text = {
            Column {
                OutlinedTextField(
                    value = name,
                    onValueChange = { name = it },
                    label = { Text("Name") },
                    singleLine = true,
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 8.dp)
                )

                OutlinedTextField(
                    value = email,
                    onValueChange = { email = it },
                    label = { Text("Email") },
                    singleLine = true,
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 8.dp),
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email)
                )

                OutlinedTextField(
                    value = password,
                    onValueChange = { password = it },
                    label = { Text("Password") },
                    singleLine = true,
                    visualTransformation = if (passwordVisible) VisualTransformation.None else PasswordVisualTransformation(),
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 8.dp),
                    trailingIcon = {
                        IconButton(onClick = { passwordVisible = !passwordVisible }) {
                            Icon(
                                imageVector = if (passwordVisible) Icons.Default.Visibility else Icons.Default.VisibilityOff,
                                contentDescription = if (passwordVisible) "Hide password" else "Show password"
                            )
                        }
                    }
                )
            }
        },
        confirmButton = {
            Button(
                onClick = { onSignUp(name, email, password) },
                enabled = name.isNotBlank() && email.isNotBlank() && password.isNotBlank(),
                colors = ButtonDefaults.buttonColors(
                    containerColor = AppColors.Primary
                )
            ) {
                Text("Create Account")
            }
        },
        dismissButton = {
            OutlinedButton(onClick = onDismiss) {
                Text("Cancel")
            }
        }
    )
}

@Composable
fun SettingsItem(
    icon: ImageVector,
    title: String,
    onClick: () -> Unit,
    textColor: Color = AppColors.TextPrimary
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = textColor,
            modifier = Modifier.size(24.dp)
        )

        Spacer(modifier = Modifier.width(16.dp))

        Text(
            text = title,
            fontSize = 16.sp,
            color = textColor,
            modifier = Modifier.weight(1f)
        )

        Icon(
            imageVector = Icons.Default.ChevronRight,
            contentDescription = null,
            tint = AppColors.Gray,
            modifier = Modifier.size(20.dp)
        )
    }
}

File: iosApp/CocktailCraft/CocktailCraft/Screens/ProfileScreen.swift
import SwiftUI
import shared

struct ProfileScreen: View {
    @StateObject private var profileViewModel = ViewModelProvider.shared.profileViewModel
    @StateObject private var themeViewModel = ViewModelProvider.shared.themeViewModel
    @EnvironmentObject var navigationCoordinator: NavigationCoordinator
    
    @State private var showLogoutAlert = false
    @State private var showSignInDialog = false
    @State private var showSignUpDialog = false
    @State private var showErrorAlert = false
    
    private var userName: String {
        profileViewModel.user?.name ?? "Guest User"
    }
    
    private var userEmail: String {
        profileViewModel.user?.email ?? "guest@example.com"
    }
    
    var body: some View {
        NavigationView {
            contentView
                .navigationTitle("Profile")
                .navigationBarTitleDisplayMode(.large)
                .alert("Logout", isPresented: $showLogoutAlert) {
                    Button("Cancel", role: .cancel) { }
                    Button("Logout", role: .destructive) {
                        profileViewModel.signOut()
                        navigationCoordinator.navigateToTab(.home)
                    }
                } message: {
                    Text("Are you sure you want to logout?")
                }
                .alert("Error", isPresented: $showErrorAlert) {
                    Button("OK") { }
                } message: {
                    Text("An error occurred")
                }
                .sheet(isPresented: $showSignInDialog) {
                    signInSheet
                }
                .sheet(isPresented: $showSignUpDialog) {
                    signUpSheet
                }
        }
    }
    
    @ViewBuilder
    private var contentView: some View {
        ZStack {
            mainScrollView
            if profileViewModel.isLoading {
                loadingOverlay
            }
        }
    }
    
    private var mainScrollView: some View {
        ScrollView {
            VStack(spacing: 16) {
                profileHeader
                if profileViewModel.isSignedIn {
                    accountSettingsSection
                }
                appSettingsSection
                aboutSection
                Spacer(minLength: 20)
            }
            .padding(.vertical)
        }
    }
    
    private var profileHeader: some View {
        ProfileCard(
            userName: userName,
            userEmail: userEmail,
            isSignedIn: profileViewModel.isSignedIn,
            onSignIn: { showSignInDialog = true },
            onSignUp: { showSignUpDialog = true }
        )
        .padding(.horizontal)
    }
    
    private var accountSettingsSection: some View {
        SettingsSection(title: "Account Settings") {
            VStack(spacing: 0) {
                accountSettingsRows
            }
        }
        .padding(.horizontal)
    }
    
    private var accountSettingsRows: some View {
        Group {
            SettingsRow(
                icon: "person",
                title: "Edit Profile",
                action: { /* Handle edit profile */ }
            )
            
            Divider()
            
            SettingsRow(
                icon: "lock",
                title: "Change Password",
                action: { /* Handle change password */ }
            )
            
            Divider()
            
            SettingsRow(
                icon: "envelope",
                title: "Email Preferences",
                action: { /* Handle email preferences */ }
            )
            
            Divider()
            
            SettingsRow(
                icon: "bell",
                title: "Notification Settings",
                action: { /* Handle notification settings */ }
            )
        }
    }
    
    private var appSettingsSection: some View {
        SettingsSection(title: "App Settings") {
            VStack(spacing: 0) {
                appSettingsRows
                if profileViewModel.isSignedIn {
                    Divider()
                    logoutRow
                }
            }
        }
        .padding(.horizontal)
    }
    
    private var appSettingsRows: some View {
        Group {
            SettingsRow(
                icon: "calendar",
                title: "Order History",
                action: {
                    navigationCoordinator.navigateToTab(.orders)
                }
            )
            
            Divider()
            
            SettingsRow(
                icon: "questionmark.circle",
                title: "Help & Support",
                action: { /* Handle help & support */ }
            )
            
            Divider()
            
            SettingsRow(
                icon: "icloud.slash",
                title: "Offline Mode",
                action: {
                    navigationCoordinator.navigateToOfflineMode()
                }
            )
            
            Divider()
            
            themeSettingsRows
        }
    }
    
    private var themeSettingsRows: some View {
        Group {
            SettingsToggleRow(
                icon: "gear",
                title: "Follow System Theme",
                subtitle: "Automatic",
                isOn: Binding(
                    get: { false }, // Simplified for now
                    set: { _ in }
                )
            )
            
            Divider()
            
            SettingsToggleRow(
                icon: themeViewModel.isDarkMode ? "moon.fill" : "sun.max",
                title: "Dark Mode",
                subtitle: themeViewModel.isDarkMode ? "On" : "Off",
                isOn: Binding(
                    get: { themeViewModel.isDarkMode },
                    set: { _ in 
                        themeViewModel.setDarkMode(enabled: !themeViewModel.isDarkMode)
                    }
                ),
                isEnabled: true
            )
        }
    }
    
    private var logoutRow: some View {
        SettingsRow(
            icon: "rectangle.portrait.and.arrow.right",
            title: "Logout",
            textColor: .red,
            action: { showLogoutAlert = true }
        )
    }
    
    private var aboutSection: some View {
        SettingsSection(title: "About") {
            VStack(alignment: .leading, spacing: 8) {
                Text("CocktailCraft")
                    .font(.system(size: 16, weight: .medium))
                
                Text("Version 1.0.0")
                    .font(.system(size: 14))
                    .foregroundColor(.secondary)
                
                Text("© 2024 CocktailCraft. All rights reserved.")
                    .font(.system(size: 12))
                    .foregroundColor(.secondary)
                    .padding(.top, 8)
            }
        }
        .padding(.horizontal)
    }
    
    private var loadingOverlay: some View {
        Color.black.opacity(0.5)
            .ignoresSafeArea()
            .overlay(
                ProgressView()
                    .scaleEffect(1.5)
                    .progressViewStyle(CircularProgressViewStyle(tint: .white))
            )
    }
    
    private var signInSheet: some View {
        SignInDialog(
            isPresented: $showSignInDialog,
            onSignIn: { email, password in
                profileViewModel.signIn(email: email, password: password)
            }
        )
    }
    
    private var signUpSheet: some View {
        SignUpDialog(
            isPresented: $showSignUpDialog,
            onSignUp: { name, email, password in
                // Note: signUp method might not exist yet in ProfileViewModel
                // For now, just close the dialog
                showSignUpDialog = false
            }
        )
    }
}